modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/render.tsx

```diff
@@ -47,10 +47,10 @@ export default function render(
 	},
 	container: Element
 ) {
-	if (!(window.Liferay as any).SPA || (window.Liferay as any).SPA.app) {
+	if (!Liferay.SPA || Liferay.SPA.app) {
```

Comment:
We should definitely provide typings for `Liferay` so that people can use it as documentation :+1: 

When I did the sample Angular project for the JS Toolkit, I had to use `declare Liferay: any` because we were lacking that, but it was clear that publishing the types for our `Liferay` interface would be a win (at least for the public parts).